silkrpc/http/reply.hpp

```diff
@@ -19,28 +19,34 @@
 
 namespace silkrpc::http {
 
+/// The status of the reply.
+enum StatusType {
+    processing_continue = 100,
+    ok = 200,
+    created = 201,
+    accepted = 202,
+    no_content = 204,
+    multiple_choices = 300,
+    moved_permanently = 301,
+    moved_temporarily = 302,
+    not_modified = 304,
+    bad_request = 400,
+    unauthorized = 401,
+    forbidden = 403,
+    not_found = 404,
+    internal_server_error = 500,
+    not_implemented = 501,
+    bad_gateway = 502,
+    service_unavailable = 503
+};
```

Comment:
The reason is due to stream introduction: `StatusType` should be accessed also when `Reply` is not used, so I moved it out `Reply` and declared `boost::asio::const_buffer to_buffer(StatusType status)` in .hpp to use is in json stream.
It will be clear on next PR.